lib/core/utils/flattened-tree.js

```diff
@@ -51,6 +51,20 @@ function getSlotChildren(node) {
 	return retVal;
 }
 
+const possibleShadowRoots = ['article', 'aside', 'blockquote', 
```

Comment:
What about custom elements? Could we do a blacklist approach instead of a whitelist and exclude `html`?

Reply:
Custom elements are allowed as long as they have a dash in the name. Chrome doesn't let you attach shadow trees to other stuff, so I think this works best.

Reply:
@WilcoFiers could you add a test to make sure that that statement does not change?

Reply:
You mean test that something in the codebase didn't modify this array somehow? I could freeze it if you want.

Reply:
Maybe write a test for custom elements?

Reply:
@WilcoFiers No, what I was suggesting is that we write a test to make sure you cannot attach a shadow root to a custom element without a hyphen